torch/csrc/jit/script/python_sugared_value.cpp

```diff
@@ -302,6 +305,42 @@ std::shared_ptr<SugaredValue> ModuleValue::attr(
     auto list = g.insertNode(g.createTuple(params))->output();
     return std::make_shared<ConstantParameterList>(list);
   }
+
+  // If recursive script mode is on, create a ScriptModule and register it as
+  // as submodule or register a python method as a script::Method
+  if (getRecursiveScriptMode()) {
+    if (py::isinstance(attr, py::module::import("torch.nn").attr("Module"))) {
+      // If the module is a submodule of the py_module, convert it to a
+      // ScriptModule and add it as a submodule to the script::Module. This
+      // enables lazy strong-ification of modules.
+      auto result =
+          py::module::import("torch.jit")
+              .attr("_make_strong_submodule")(field, attr, py_module_);
+      if (!result.is_none()) {
+        auto submodule = as_module(result);
+        AT_CHECK(
+            submodule,
+            "Result of torch.jit._make_strong_submodule "
+            "was not a ScriptModule");
+        // The module was a submodule of the nn.Module, so register it here
+        // and return the submodule.
+        module_->register_module(field, submodule);
+        auto v = module_->find_module(field);
+        return std::make_shared<ModuleValue>(
+            m.graph()->insertGetAttr(self_, field), v, attr);
+      }
+    } else if (py::isinstance<py::function>(attr)) {
```

Comment:
We need to write a follow up task to improve the error messages where a error occurs nested through this code.